pkg/controller/promotionreconciler/reconciler.go

```diff
@@ -239,18 +229,18 @@ func refForIST(ist *imagev1.ImageStreamTag) (*branchReference, error) {
 	commit := metadata.Config.Labels["io.openshift.build.commit.id"]
 	sourceLocation := metadata.Config.Labels["io.openshift.build.source-location"]
 	if branch == "" {
-		return nil, nonRetriableError{errors.New("imageStreamTag has no `io.openshift.build.commit.ref` label, can't find out source branch")}
+		return nil, controllerutil.TerminalError(errors.New("imageStreamTag has no `io.openshift.build.commit.ref` label, can't find out source branch"))
 	}
 	if commit == "" {
-		return nil, nonRetriableError{errors.New("ImageStreamTag has no `io.openshift.build.commit.id` label, can't find out source commit")}
+		return nil, controllerutil.TerminalError(errors.New("ImageStreamTag has no `io.openshift.build.commit.id` label, can't find out source commit"))
```

Comment:
This could/should be split out of this PR?

Reply:
https://github.com/openshift/ci-tools/pull/830